### PR TITLE
Maps: The Return

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: include/si.h [hunk 1]
```diff
@@ -16,7 +16,7 @@ typedef s32 ScriptID;
 #define SI_SAVE_VAR(v) (v - 170000000)
 #define SI_ARRAY(v) (v - 190000000)
 #define SI_ARRAY_FLAG(v) (v - 210000000)
-#define SI_FIXED(v) ((s32)(v * 1024.0f) - 230000000) // See float_to_fixed_var
+#define SI_FIXED(v) ((v * 1024.0f) + -230000000) // See float_to_fixed_var
```
Comment (nanaian) [inline include/si.h:19]: wat

File: src/world/area_kmr/kmr_03/8C83A0.c [hunk 2]
```diff
@@ -2,3 +2,336 @@
 
 #include "world/common/SomeMatrixOperations.inc.c"
 
+// 8C9780
```
Comment (nanaian) [inline src/world/area_kmr/kmr_03/8C83A0.c:5]: Drop this comment

File: src/world/area_kmr/kmr_03/8C8140.c [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+    0x00000000, 0x00000000, 0x00000000, 0x001A0063,
+};
+
+s32 M(npcGroupList_80241450)[] = {
+    0x00000001, M(npcGroup_80241260), 0x00020000, 0x00000000, 0x00000000, 0x00000000,
+};
+
+s32 padding2[] = {0, 0};
+
+// 8C9400
```
Comment (nanaian) [inline src/world/area_kmr/kmr_03/8C8140.c:257]: New file should probably be here, given the padding.

File: src/world/area_kmr/kmr_03/8C8140.c [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+    SI_END_IF(),
+    SI_WAIT_FRAMES(1),
+    SI_RETURN(),
+    SI_END(),
+};
+// *INDENT-ON*
+
+s32 padding[] = {0, 0};
+
+// 8C88E0
```
Comment (nanaian) [inline src/world/area_kmr/kmr_03/8C8140.c:75]: New file

File: src/world/area_kmr/kmr_03/8C83A0.c [hunk 5]
```diff
@@ -2,3 +2,336 @@
...
+    SI_BIND(M(script_ShakeTree_80241B50), TriggerFlag_WALL_HAMMER, 52, NULL),
+    SI_BIND(M(script_ShakeTree_80241B50), TriggerFlag_BOMB, M(triggerCoord_802422A8), NULL),
+    SI_RETURN(),
+    SI_END(),
+};
+
+s32 padding3[] = {0};
+
+// 8CA2D0
+// *INDENT-OFF*
```
Comment (nanaian) [inline src/world/area_kmr/kmr_03/8C83A0.c:251]: New file

File: src/world/area_sam/sam_11/D3A930.c [hunk 6]
```diff
@@ -4,13 +4,20 @@ INCLUDE_ASM(s32, "world/area_sam/sam_11/D3A930", func_80240360_D3A930);
 
 INCLUDE_ASM(s32, "world/area_sam/sam_11/D3A930", func_8024052C_D3AAFC);
 
-INCLUDE_ASM(s32, "world/area_sam/sam_11/D3A930", func_80240540_D3AB10);
+// Almost the same as Set80151310, but we're setting it to 2 in the first func instead of 1
```
Comment (nanaian) [inline src/world/area_sam/sam_11/D3A930.c:7]: Comment seems redundant

File: src/world/common/NotDoesScriptExist.inc.c [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#include "common.h"
+
+static ApiStatus NotDoesScriptExist(ScriptInstance* script, s32 isInitialCall) {
```
Comment (nanaian) [inline src/world/common/NotDoesScriptExist.inc.c:3]: (This is documentation, but I've noticed it now:) this should be called `AwaitScriptComplete`

**Review by nanaian — CHANGES_REQUESTED** [review]

File: include/si.h (hunk 1 above)
Comment (ethteck) [reply include/si.h:19]: Changing to this allows me to call SI_FIXED and match a function, but I can just leave this alone and not use it in that function if you prefer.

Comment (nanaian) [reply include/si.h:19]: No, it's fine, I was just wondering why

Comment (ethteck) [reply include/si.h:19]: gorcha

**Review by nanaian — APPROVED** [review]

**Review by pixel-stuck — APPROVED** [review]

**Pull request merged**